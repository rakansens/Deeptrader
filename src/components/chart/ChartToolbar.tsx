"use client";

import { ToggleGroup, ToggleGroupItem } from "@/components/ui/toggle-group";
import { Switch } from "@/components/ui/switch";
import { Label } from "@/components/ui/label";
import { ThemeToggle } from "@/components/ui/theme-toggle";

interface ChartToolbarProps {
  timeframe: string;
  onTimeframeChange: (timeframe: string) => void;
  symbol?: string;
  onSymbolChange?: (symbol: string) => void;
  indicators: {
    ma: boolean;
    rsi: boolean;
    macd?: boolean;
    boll?: boolean;
  };
  onIndicatorsChange: (value: {
    ma: boolean;
    rsi: boolean;
    macd?: boolean;
    boll?: boolean;
  }) => void;
}

const TIMEFRAMES = ["1m", "5m", "15m", "1h", "4h", "1d"];
const SYMBOLS = [
  { value: "BTCUSDT", label: "BTC/USDT" },
  { value: "ETHUSDT", label: "ETH/USDT" },
  { value: "BNBUSDT", label: "BNB/USDT" },
];

export default function ChartToolbar({
  timeframe,
  onTimeframeChange,
  symbol = "BTCUSDT",
  onSymbolChange,
  indicators,
  onIndicatorsChange,
}: ChartToolbarProps) {
  return (
    <div className="flex flex-col gap-2 p-2 bg-background border-b border-border md:flex-row md:items-center md:justify-between">
      <div className="flex gap-3 items-center">
        {onSymbolChange && (
          <select
            value={symbol}
            onChange={(e) => onSymbolChange(e.target.value)}
            className="border rounded-md px-2 py-1 text-sm bg-background"
          >
            {SYMBOLS.map((s) => (
              <option key={s.value} value={s.value}>
                {s.label}
              </option>
            ))}
          </select>
        )}
        <ToggleGroup
          type="single"
          value={timeframe}
          onValueChange={(v) => v && onTimeframeChange(v)}
        >
          {TIMEFRAMES.map((tf) => (
            <ToggleGroupItem key={tf} value={tf} aria-label={`Timeframe ${tf}`}>
              {tf}
            </ToggleGroupItem>
          ))}
        </ToggleGroup>
      </div>
      <div className="flex items-center gap-3">
        <div className="flex items-center gap-2">
          <Switch
            id="ma-toggle"
            checked={indicators.ma}
            onCheckedChange={(v) =>
              onIndicatorsChange({ ...indicators, ma: v })
            }
          />
          <Label htmlFor="ma-toggle">MA</Label>
        </div>
        <div className="flex items-center gap-2">
          <Switch
            id="rsi-toggle"
            checked={indicators.rsi}
            onCheckedChange={(v) =>
              onIndicatorsChange({ ...indicators, rsi: v })
            }
          />
          <Label htmlFor="rsi-toggle">RSI</Label>
        </div>
        <div className="flex items-center gap-2">
          <Switch
            id="macd-toggle"
            checked={!!indicators.macd}
            onCheckedChange={(v) =>
              onIndicatorsChange({ ...indicators, macd: v })
            }
          />
          <Label htmlFor="macd-toggle">MACD</Label>
        </div>
        <div className="flex items-center gap-2">
          <Switch
            id="boll-toggle"
            checked={!!indicators.boll}
            onCheckedChange={(v) =>
              onIndicatorsChange({ ...indicators, boll: v })
            }
          />
          <Label htmlFor="boll-toggle">BOLL</Label>
        </div>
        <ThemeToggle />
      </div>
    </div>
  );
}
