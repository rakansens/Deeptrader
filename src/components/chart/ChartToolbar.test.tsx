import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import ChartToolbar from './ChartToolbar'

describe('ChartToolbar', () => {
  it('handles interactions', async () => {
    const user = userEvent.setup()
    const onTf = jest.fn()
    const onInd = jest.fn()
    render(
      <ChartToolbar
        timeframe="1m"
        onTimeframeChange={onTf}
        indicators={{ ma: false, rsi: false, macd: false, boll: false }}
        onIndicatorsChange={onInd}
      />
    )

    await user.click(screen.getByRole('button', { name: '5m' }))
    expect(onTf).toHaveBeenCalledWith('5m')

    await user.click(screen.getByLabelText('MA'))
    expect(onInd).toHaveBeenLastCalledWith({ ma: true, rsi: false, macd: false, boll: false })
  })
})
