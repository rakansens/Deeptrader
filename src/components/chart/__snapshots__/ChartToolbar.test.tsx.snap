// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartToolbar matches snapshot 1`] = `
<div
  class="flex flex-col gap-2 p-2 bg-background border-b border-border md:flex-row md:items-center md:justify-between"
>
  <div
    class="flex gap-3 items-center"
  >
    <div
      class="flex items-center justify-center gap-1"
      dir="ltr"
      role="group"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-checked="true"
        aria-label="Timeframe 1m"
        class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors hover:bg-muted hover:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=on]:bg-accent data-[state=on]:text-accent-foreground bg-transparent h-10 px-3"
        data-radix-collection-item=""
        data-state="on"
        role="radio"
        tabindex="-1"
        type="button"
      >
        1m
      </button>
      <button
        aria-checked="false"
        aria-label="Timeframe 5m"
        class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors hover:bg-muted hover:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=on]:bg-accent data-[state=on]:text-accent-foreground bg-transparent h-10 px-3"
        data-radix-collection-item=""
        data-state="off"
        role="radio"
        tabindex="-1"
        type="button"
      >
        5m
      </button>
      <button
        aria-checked="false"
        aria-label="Timeframe 15m"
        class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors hover:bg-muted hover:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=on]:bg-accent data-[state=on]:text-accent-foreground bg-transparent h-10 px-3"
        data-radix-collection-item=""
        data-state="off"
        role="radio"
        tabindex="-1"
        type="button"
      >
        15m
      </button>
      <button
        aria-checked="false"
        aria-label="Timeframe 1h"
        class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors hover:bg-muted hover:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=on]:bg-accent data-[state=on]:text-accent-foreground bg-transparent h-10 px-3"
        data-radix-collection-item=""
        data-state="off"
        role="radio"
        tabindex="-1"
        type="button"
      >
        1h
      </button>
      <button
        aria-checked="false"
        aria-label="Timeframe 4h"
        class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors hover:bg-muted hover:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=on]:bg-accent data-[state=on]:text-accent-foreground bg-transparent h-10 px-3"
        data-radix-collection-item=""
        data-state="off"
        role="radio"
        tabindex="-1"
        type="button"
      >
        4h
      </button>
      <button
        aria-checked="false"
        aria-label="Timeframe 1d"
        class="inline-flex items-center justify-center rounded-md text-sm font-medium ring-offset-background transition-colors hover:bg-muted hover:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=on]:bg-accent data-[state=on]:text-accent-foreground bg-transparent h-10 px-3"
        data-radix-collection-item=""
        data-state="off"
        role="radio"
        tabindex="-1"
        type="button"
      >
        1d
      </button>
    </div>
  </div>
  <div
    class="flex items-center gap-3"
  >
    <div
      class="flex items-center gap-2"
    >
      <button
        aria-checked="false"
        class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
        data-state="unchecked"
        id="ma-toggle"
        role="switch"
        type="button"
        value="on"
      >
        <span
          class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
          data-state="unchecked"
        />
      </button>
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        for="ma-toggle"
      >
        MA
      </label>
    </div>
    <div
      class="flex items-center gap-2"
    >
      <button
        aria-checked="false"
        class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
        data-state="unchecked"
        id="rsi-toggle"
        role="switch"
        type="button"
        value="on"
      >
        <span
          class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
          data-state="unchecked"
        />
      </button>
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        for="rsi-toggle"
      >
        RSI
      </label>
    </div>
    <div
      class="flex items-center gap-2"
    >
      <button
        aria-checked="false"
        class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
        data-state="unchecked"
        id="macd-toggle"
        role="switch"
        type="button"
        value="on"
      >
        <span
          class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
          data-state="unchecked"
        />
      </button>
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        for="macd-toggle"
      >
        MACD
      </label>
    </div>
    <div
      class="flex items-center gap-2"
    >
      <button
        aria-checked="false"
        class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
        data-state="unchecked"
        id="boll-toggle"
        role="switch"
        type="button"
        value="on"
      >
        <span
          class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
          data-state="unchecked"
        />
      </button>
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        for="boll-toggle"
      >
        BOLL
      </label>
    </div>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 w-10"
      data-state="closed"
      id="radix-:re:"
      type="button"
    >
      <svg
        class="lucide lucide-sun h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="4"
        />
        <path
          d="M12 2v2"
        />
        <path
          d="M12 20v2"
        />
        <path
          d="m4.93 4.93 1.41 1.41"
        />
        <path
          d="m17.66 17.66 1.41 1.41"
        />
        <path
          d="M2 12h2"
        />
        <path
          d="M20 12h2"
        />
        <path
          d="m6.34 17.66-1.41 1.41"
        />
        <path
          d="m19.07 4.93-1.41 1.41"
        />
      </svg>
      <svg
        class="lucide lucide-moon absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
        />
      </svg>
      <span
        class="sr-only"
      >
        テーマを切り替え
      </span>
    </button>
  </div>
</div>
`;
